app/services/queue_updater.rb

```diff
@@ -7,10 +7,15 @@
 # cases such as when two similar contacts enter the queue (almost) simultaneously or if the time_in_status returned by OC SOAP service
 # is not reliable (for example due to lag). The class will attempt to account for these cases but does not guarantee 100% accuracy.
 class QueueUpdater
+  include Now
   # Current_time parameter should ALWAYS be current time (Time.zone.now), except for tests
   # Last_success should contain a timestamp for when the update job was previously run successfully
   def initialize(current_time, last_success)
-    @current_time = current_time
+    @current_time = if !current_time.nil?
```

Comment:
Could be written as `current_time || now` but I don't know if it's better, clearer or idiomatic Ruby...